pywbem/_recorder.py

```diff
@@ -618,11 +618,9 @@ def format_result(ret, max_len):
             # Format the 'summary' and 'paths' detail_levels
             if self.api_detail_level == 'summary':  # pylint: disable=R1705
                 if isinstance(ret, list):
-                    if ret:
-                        ret_type = type(ret[0]).__name__ if ret else ""
-                        return _format("list of {0}; count={1}",
-                                       ret_type, len(ret))
-                    return "Empty"
+                    ret_type = type(ret[0]).__name__ if ret else ""
+                    return _format("list of {0}; count={1}",
+                                   ret_type, len(ret))
```

Comment:
While the change does not harm (it returns the "list of ..." text with count 0 in case of an empty list instead of "Empty"), I would like to point out that the coverity tool's explanation: "If the expression isinstance(ret, list) evaluates to True, the expression if ret: will evaluate to False causing those lines to
to be dead." is incorrect. If "ret" is a list, "if ret" is True if the list is non-empty, and otherwise False.

Reply:
Yes, you're correct

Reply:
I made  slight modification to your proposed change, as I think this this makes the intention more obvious ...

```python
                if isinstance(ret, list):
                    if len(ret) > 0:
                        ret_type = type(ret[0]).__name__
                        return _format("list of {0}; count={1}",
                                       ret_type, len(ret))
                    return "Empty"
```